### PR TITLE
[BF-78] FEAT: Implement language selector UI and functionality

### PR DESCRIPTION
**Pull request by gor017**
### What are you trying to accomplish?
[BF-78]
### Testing?
image

### CODE REVIEW

**Review by rikarsen — CHANGES_REQUESTED** [review]
image
you have errors

**Comment by gor017** [comment]
> image
> 
> you have errors

Done

File: src/components/Icons/lang/EnIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React, { FC } from 'react'
+import IconProps from '@allTypes/svg-icon'
+import clsxMerge from '@lib/clsxm'
+
+export const EnIcon: FC<IconProps> = ({ fill = 'fill-[#212121]', className, ...rest }) => {
```
Comment (ArthurO2002) [inline src/components/Icons/lang/EnIcon.tsx:5]: Should be color  from paleete

File: src/components/Icons/lang/RuIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React, { FC } from 'react'
+import IconProps from '@allTypes/svg-icon'
+import clsxMerge from '@lib/clsxm'
+
+export const RuIcon: FC<IconProps> = ({ fill = 'fill-[#212121]', className, ...rest }) => {
```
Comment (ArthurO2002) [inline src/components/Icons/lang/RuIcon.tsx:5]: Here too

File: src/pages/_app.tsx [hunk 3]
```diff
@@ -27,6 +31,14 @@ interface IAppProps {
 const App = ({ Component, pageProps }: IAppProps) => {
   const HierarchicalLayout = Layouts[Component.Layout]
 
+  useEffect(() => {
+    const data = window.localStorage.getItem('language')
+
+    if (data !== null) {
+      dispatch(usersMiddleware.changeLanguage(data))
+    }
```
Comment (ArthurO2002) [inline src/pages/_app.tsx:39]:
```suggestion
    if (data) {
      dispatch(usersMiddleware.changeLanguage(data))
    }
```

File: src/redux/slices/users/middleware.ts [hunk 4]
```diff
@@ -72,10 +81,27 @@ const register = (params: ISignUpParams) => async (dispatch: AppDispatch) => {
...
+  try {
+    dispatch(setLanguageChangeLoading(true))
+    dispatch(setLanguage(lng))
+
+    i18n.changeLanguage(lng)
+    localStorage.setItem('language', lng)
+
+    dispatch(setError(null))
+  } catch (error) {
+    dispatch(setError((error as IError).response.data.status.message))
```
Comment (ArthurO2002) [inline src/redux/slices/users/middleware.ts:94]: Add ? to not brake the page

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/redux/slices/users/middleware.ts (hunk 4 above)
Comment (vahea95) [reply src/redux/slices/users/middleware.ts]: done

File: src/components/Icons/lang/RuIcon.tsx (hunk 2 above)
Comment (vahea95) [reply src/components/Icons/lang/RuIcon.tsx]: done

File: src/components/Icons/lang/EnIcon.tsx (hunk 1 above)
Comment (vahea95) [reply src/components/Icons/lang/EnIcon.tsx]: done

**Review by ArthurO2002 — APPROVED** [review]

**Pull request merged**